### ISSUE TITLE
Trigger "player killed entity" don't work with NBT data entity in Advancements Pack.

### ISSUE THREAD

**Issue by SoulxluoS** [PaperMC/Paper#4252]
I play with BlazeandCave's Advancements Pack [1.16 Datapack] last version. Advancements such as "custom_boss_fight", "trick_or_treat" etc. don't work. But on vanila all work fine. 

Code from the achievement "custom_boss_fight":

{
    "display": {
        "icon": {
            "item": "minecraft:diamond_helmet"
        },
        "title": {"translate":"Custom Boss Fight!"},
        "description": {"translate":"Give a zombie full diamond armor and a diamond sword... then kill it"},
		"frame": "challenge"
    },
    "parent": "blazeandcave:combat/freezing",
    "rewards": {
        "function": "bc_rewards:combat/custom_boss_fight",
		"experience": 100
    },		
    "criteria": {
        "zombie": {
            "trigger": "minecraft:player_killed_entity",
            "conditions": {
                "entity": {
                    "type": "zombie",
		"nbt": "{HandItems:[{id:\"minecraft:diamond_sword\",Count:1b}],ArmorItems:[{id:\"minecraft:diamond_boots\",Count:1b},{id:\"minecraft:diamond_leggings\",Count:1b},{id:\"minecraft:diamond_chestplate\",Count:1b},{id:\"minecraft:diamond_helmet\",Count:1b}]}"
                }
            }
        }
    }
}

### Steps/models to reproduce:
Installing last version BlazeandCave's Advancements Pack [https://www.planetminecraft.com/data-pack/blazeandcave-s-advancements-pack-1-12/](url). Try completing achievements such as "custom_boss_fight", "trick_or_treat" on Paper server and Vanilla server.

### Plugin list:
Not plugins

### Paper version:
1.16.2 162 - 1.16.5 471

**Comment by rrvk** [comment]
Last i checked this was still a thing, but i will check if i have time after working

**Comment by SoulxluoS** [comment]
Confirm 1.16.4 - 320. I can't give you any more information

**Comment by Machine-Maker** [comment]
I can't replicate this. I'll walk you through my exact steps. I created a datapack with only one advancement, the root one (shown below). I then make sure I did not have the advancement by getting rid of all my advancements (`/advancement revoke Machine_Maker everything`). Then I spawned in a zombie with the command (found below) and killed it. I got the achievement.

#### Root advancement
```json
{
    "display": {
        "title": {
            "text": "Killed Zombie",
            "color": "dark_green",
            "bold": false
        },
        "description": {
            "text": "Kill a zombie",
            "underlined": true
        },
        "icon": {
            "item": "minecraft:zombie_head"
        },
        "frame": "task",
        "show_toast": true,
        "announce_to_chat": true,
        "hidden": false,
        "background": "minecraft:textures/gui/advancements/backgrounds/stone.png"
    },
    "criteria": {
        "zombie": {
            "trigger": "minecraft:player_killed_entity",
            "conditions": {
                "entity": {
                    "type": "minecraft:zombie",
                    "nbt": "{HandItems:[{id:\"minecraft:diamond_sword\",Count:1b}],ArmorItems:[{id:\"minecraft:diamond_boots\",Count:1b},{id:\"minecraft:diamond_leggings\",Count:1b},{id:\"minecraft:diamond_chestplate\",Count:1b},{id:\"minecraft:diamond_helmet\",Count:1b}]}"
                }
            }
        }
    }
}
```

#### Zombie Summon Command
Too long for normal chat bar so I used a command block.
`summon minecraft:zombie -27.14 94 -12.51 {NoAI:1b,HandItems:[{id:"minecraft:diamond_sword",Count:1b}],ArmorItems:[{id:"minecraft:diamond_boots",Count:1b},{id:"minecraft:diamond_leggings",Count:1b},{id:"minecraft:diamond_chestplate",Count:1b},{id:"minecraft:diamond_helmet",Count:1b}]}`

**Comment by SoulxluoS** [comment]
But if summon zombie without equipment and give him armor and sword then not work.
1.16.5 Paper-471

**Comment by Machine-Maker** [comment]
Chances are, that this is actually fixed by #5740 and is also probably a duplicate of #3729 since that advancement uses the player_killed_entity trigger with custom NBT for killing a raid captain.
